GooglePlacesAutocomplete.js

```diff
@@ -302,7 +330,7 @@ export default class GooglePlacesAutocomplete extends Component {
 
       request.open(
         'GET',
-        'https://maps.googleapis.com/maps/api/place/details/json?' +
+        `${this.state.url}/place/details/json?` +
```

Comment:
Maybe I'm reading this wrong -- shouldn't this be:

`${this.state.url}`

Adding `/place/details/json` locks you into the google maps request format. This occurs in a few other places as well. 

Unless I'm missing something, i'd suggest something like:

```{this.props.requestUrl ? `${this.state.url}` : `${this.state.url}/place/details/json?`} ```

Reply:
On second thought, maybe this should be handled in the getRequestUrl function. I don't love checking against props with every request.

How about a big switch / case or something like that?

Reply:
Great point. 

I was very back and forth on this. At this point, although I haven't dug into the code enough to say for certain, but I believe the code to render the results is tightly coupled to the google api.  Therefore, for all intents and purposes, this only supports using a proxy server of the Google API for now.

Reply:
That makes sense. Might be worth making that explicit in the docs -- any proxy server needs to expose the exact endpoints that are getting called or else someone could end up with a nasty stack trace. 

May also be worthwhile to add a verbose error if a non-google url is passed (e.g., regex the url string, if it's not a valid google places api url then throw). The title of the repo should deter people..... but i wouldn't count on it.

Reply:
I will make it clear in the doc.  

> regex the url string, if it's not a valid google places api url then throw

If someone creates a proper proxy server within their own api, it might not have the word google in it. This is really hard to solve.  I feel like this is becoming a whole error handling library for people who don't read the docs. 

I am going to release as is. I can always iterate on this. 

Thanks for the review!